telemetry/data_feature_usage.go

```diff
@@ -48,21 +77,49 @@ func getFeatureUsage(ctx sessionctx.Context) (*featureUsage, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	defer func() {
+		if r := recover(); r != nil {
+			switch x := r.(type) {
+			case string:
+				err = errors.New(x)
+			case error:
+				err = x
+			default:
+				err = errors.New("unknown failure")
+			}
+		}
+	}()
+	infoSchema := ctx.GetSessionVars().GetInfoSchema().(infoschema.InfoSchema)
+
+	// check ClusterIndex information for each table
+	// row: 0 = table_name_hash, 1 = TIDB_PK_TYPE, 2 = TABLE_SCHEMA (db), 3 = TABLE_NAME
+
 	for _, row := range rows {
-		if row.Len() < 2 {
+		if row.Len() < 4 {
 			continue
 		}
-		isClustered := false
+		tblClusteredInfo := TableClusteredInfo{false, "NA"}
 		if row.GetString(1) == "CLUSTERED" {
-			isClustered = true
+			tblClusteredInfo.IsClustered = true
+			table, err := infoSchema.TableByName(model.NewCIStr(row.GetString(2)), model.NewCIStr(row.GetString(3)))
+			if err != nil {
+				continue
+			}
+			tableInfo := table.Meta()
+			if tableInfo.PKIsHandle {
+				tblClusteredInfo.ClusterPKType = "INT"
+			} else if tableInfo.IsCommonHandle {
+				tblClusteredInfo.ClusterPKType = "NON_INT"
```

Comment:
Is it possible to query this information from the information_schema?

/cc @lzmhhh123  @lysu PTAL

Reply:
It doesn't matter. The SQL accessing `information_schema` reads from information schema too.

Reply:
+1 information_schema has hidden int detail for the user as PM's design, so we only can read it from tableInfo